radicle-cli/src/commands/patch.rs

```diff
@@ -67,14 +68,14 @@ pub enum Operation {
         message: Message,
     },
     Show {
-        patch_id: PatchId,
+        patch_id: String,
     },
     Update {
-        patch_id: Option<PatchId>,
+        patch_id: Option<String>,
         message: Message,
     },
     Checkout {
-        patch_id: PatchId,
+        patch_id: String,
```

Comment:
Would it be possible to leave these as `PatchId`, and do the resolution before creating this object?

Reply:
You would need to be able to have the repository to do that, which in turn requires an `Id` for knowing which repository you want to search for in case it's a short id.

Reply:
Same in the other places where we changed to String. The parsing/resolution should happen earlier and we should keep those structs typed properly.

Reply:
Hmm true, how about having a type like `Rev` that has to be a valid git rev or sha-1 prefix with a minimum length. That way we at least catch malformed strings early. I'm thinking this also makes it possible to use refs to refer to issues :thinking:

Reply:
we can resolve the is as before and then call the as_object_id (a better name can be normalize_object_id), in this way we keep the parsing args safety and we support short id.

Now that I'm thinking to it, this solution also introduce a minimal change to the current solution